opentelemetry-sdk/src/export/trace.rs

```diff
@@ -98,3 +105,32 @@ pub struct SpanData {
     /// Instrumentation scope that produced this span
     pub instrumentation_scope: InstrumentationScope,
 }
+
+/// Errors returned by shutdown operations in the Export API.
+#[derive(Error, Debug)]
+#[non_exhaustive]
+pub enum ShutdownError {
+    /// The exporter has already been shut down.
```

Comment:
still unsure of this....I can understand that a Provider's shutdown method, if invoked more than once, can return AlreadyShutdown error. How would an Exporter's shutdown be invoked more than once, since provider protects the processor/exporter from entering that situation?

I am not even sure if processor/exporter should even have a variable like `is_shutdown` at all.. If its shutdown is invoked, it'll kill the transports etc. any further attempt to export will result in failure naturally as transports are killed.

(Sorry not really part of this PR, but this is something that I need some time to think through....)

Reply:
> I am not even sure if processor/exporter should even have a variable like is_shutdown at all.. If its shutdown is invoked, it'll kill the transports etc. any further attempt to export will result in failure naturally as transports are killed.

I feel processor should have this flag. It can't assume how the exporters are implemented. Maybe exporter's shutdown methods are just no-op, and in that case the export may continue happening if the processor doesn't have `is_shutdown` safeguard.

Reply:
What is the need for processor to have the flag? If shutdown was already signaled, then no further shutdown signal will ever arrive at the LogProcessor, as the Provider takes care of that, and provider own Processors. 

How would log processor ever get a 2nd shutdown call? Or how does it ever get an emit() call after shutdown is done?

Reply:
> How would log processor ever get a 2nd shutdown call? Or how does it ever get an emit() call after shutdown is done?

Yes correct, is_shutdown at provider level will take care of that. I missed that. However, logger::emit() as of now doesn't check for the provider flag, which needs to be fixed.

Reply:
raised a PR to discuss this further - #2462

Reply:
<FWIW>

I feel like it'd be cool if `shutdown` was idempotent - you call it, it shuts down, you call it again, it no-ops. 

I think the way i've got it here with a "AlreadyShutdown" error smells a bit and i'm happy to pick up whatever you folks decide in the other PR!

Reply:
> I feel like it'd be cool if shutdown was idempotent - you call it, it shuts down, you call it again, it no-ops.

> you call it again, it no-ops.

This maybe okay. But if a user is calling shutdown more than once, it indicates there have an bug in the way they orchestrate telemetry pipelines, so isn't it best to let them know about it...?

Reply:
Is logging it sufficient ?

Reply:
I reckon there’s a clear line there between errors that actually reflect some failure of the system, and errors that have no impact and are just hinting that the caller has done something funny.

By way of example a flush that fails become the connections gone is very different to “probably you didn’t mean to call this twice”. 

That stdout error one is interesting. Is there a story there ?

Reply:
Also and regardless - it feels like something we should have a consistent view on - when to error, when not to - that we can point at apply uniformly. That consistency is maybe even more important than what the policy itself is 🤷‍♂️